include/fs_utils.h

```diff
@@ -89,6 +89,34 @@ constexpr uint32_t OK_IF_EXISTS = 0x1;
 
 int create_dir(const std_fs::path& path, uint32_t mode, uint32_t flags = 0x0) noexcept;
 
+// Behaves like fseek, but logs an error stating the module, byte offset,
+// filename, and strerror on failure. Returns true on success and false on
+// failure. On failure, it close the file as it's no longer in a good state.
+//
+bool check_fseek(const char* module_name, const char* filename, FILE* stream,
+                 const long long offset, const int whence);
+
+// Returns a 'check_fseek' function object that behaves like the above, with
+// the same signature as fseek. This can be used in areas where lots of
+// sequential seeks are needed.
+//
+inline auto create_check_fseek(std::string module_name, std::string filename)
+{
+	auto check_fseek_lambda =
+	        [&](FILE* stream, const long long offset, const int whence) -> bool {
+		// Hold the arguments locally for the lifetime of the lamda
+		const auto local_module_name = std::move(module_name);
+		const auto local_filename    = std::move(filename);
+
+		return check_fseek(local_module_name.c_str(),
+		                   local_filename.c_str(),
+		                   stream,
+		                   offset,
+		                   whence);
+	};
+	return check_fseek_lambda;
+}
+
```

Comment:
It would be nice if you could use this mechanism to replace `fseek` and `log_layout_read_error` calls in `dos_keyboard_layout.cpp`.

Reply:
@FeralChild64 , see the last commit.

It's not as tidy as I would have liked because I had to add a bit more pass-through of the filename (to log it) as well as moving to the simpler 'FILE*' pointers.

If it's not to your liking I can drop that commit (no problems!).

Reply:
Edit.. decided to knock it out; it was too messy and introduced more file management headaches because it operates on a `FILE*` instead of the `FILE_unique_ptr`.